src/drivers/ports/esp32/uart_port.h

```diff
@@ -33,6 +33,46 @@
+#define ESP32_UART_DATA_BITS_7  (0b10 << ESP32_UART_CONF0_BIT_NUM_POS)
+#define ESP32_UART_DATA_BITS_8  (0b11 << ESP32_UART_CONF0_BIT_NUM_POS)
+#define ESP32_UART_PARITY_N     (0b00 << ESP32_UART_CONF0_PARITY_POS)
+#define ESP32_UART_PARITY_E     (0b10 << ESP32_UART_CONF0_PARITY_POS)
+#define ESP32_UART_PARITY_O     (0b11 << ESP32_UART_CONF0_PARITY_POS)
+#define ESP32_UART_STOP_BITS_1  (0b01 << ESP32_UART_CONF0_STOP_BIT_NUM_POS)
+#define ESP32_UART_STOP_BITS_15 (0b10 << ESP32_UART_CONF0_STOP_BIT_NUM_POS)
+#define UART_FORMAT_5N1   (ESP32_UART_DATA_BITS_5 | ESP32_UART_PARITY_N | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_5N15  (ESP32_UART_DATA_BITS_5 | ESP32_UART_PARITY_N | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_5O1   (ESP32_UART_DATA_BITS_5 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_5O15  (ESP32_UART_DATA_BITS_5 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_5E1   (ESP32_UART_DATA_BITS_5 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_5E15  (ESP32_UART_DATA_BITS_5 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_6N1   (ESP32_UART_DATA_BITS_6 | ESP32_UART_PARITY_N | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_6N15  (ESP32_UART_DATA_BITS_6 | ESP32_UART_PARITY_N | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_6O1   (ESP32_UART_DATA_BITS_6 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_6O15  (ESP32_UART_DATA_BITS_6 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_6E1   (ESP32_UART_DATA_BITS_6 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_6E15  (ESP32_UART_DATA_BITS_6 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_7N1   (ESP32_UART_DATA_BITS_7 | ESP32_UART_PARITY_N | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_7N15  (ESP32_UART_DATA_BITS_7 | ESP32_UART_PARITY_N | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_7O1   (ESP32_UART_DATA_BITS_7 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_7O15  (ESP32_UART_DATA_BITS_7 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_7E1   (ESP32_UART_DATA_BITS_7 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_7E15  (ESP32_UART_DATA_BITS_7 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_8N1   (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_N | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_8N15  (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_N | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_8O1   (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_8O15  (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_8E1   (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_8E15  (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_DEFAULT     UART_FORMAT_8N1
```

Comment:
But uart_init() is port-agnostic and a port is initializated with its default format. Defining UART_FORMAT_DEFAULT means that drivers know the format to their implementations.